Sources/Apollo/InterceptorRequestChain.swift

```diff
@@ -97,40 +92,73 @@ final public class InterceptorRequestChain: Cancellable, RequestChain {
     response: HTTPResponse<Operation>?,
     completion: @escaping (Result<GraphQLResult<Operation.Data>, Error>) -> Void
   ) {
-      // Empty implementation, proceedAsync(request:response:completion:interceptor:) should be
-      // used instead.
+    let nextIndex = self.currentIndex + 1
+
+    proceedAsync(
+      interceptorIndex: nextIndex,
+      request: request,
+      response: response,
+      completion: completion
+    )
   }
 
   /// Proceeds to the next interceptor in the array.
   ///
   /// - Parameters:
   ///   - request: The in-progress request object
   ///   - response: [optional] The in-progress response object, if received yet
-  ///   - completion: The completion closure to call when data has been processed and should be
-  ///   returned to the UI.
   ///   - interceptor: The interceptor that has completed processing and is ready to pass control
   ///   on to the next interceptor in the chain.
-  func proceedAsync<Operation: GraphQLOperation>(
+  ///   - completion: The completion closure to call when data has been processed and should be
+  ///   returned to the UI.
+  public func proceedAsync<Operation: GraphQLOperation>(
+    request: HTTPRequest<Operation>,
+    response: HTTPResponse<Operation>?,
+    interceptor: any ApolloInterceptor,
+    completion: @escaping (Result<GraphQLResult<Operation.Data>, Error>) -> Void
+  ) {
+    guard let currentIndex = interceptorIndexes[interceptor.id] else {
+      self.handleErrorAsync(
+        ChainError.unknownInterceptor(id: interceptor.id),
+        request: request,
+        response: response,
+        completion: completion
+      )
+      return
+    }
+
+    let nextIndex = currentIndex + 1
+
+    proceedAsync(
+      interceptorIndex: nextIndex,
+      request: request,
+      response: response,
+      completion: completion
+    )
+  }
+
+  private func proceedAsync<Operation: GraphQLOperation>(
```

Comment:
The shared logic of `proceedAsync` so the other two public functions are purely for interceptor lookup.